core/runtime/src/main/java/io/quarkus/runtime/annotations/ConfigItem.java

```diff
@@ -53,4 +53,11 @@
      * @return the default value
      */
     String defaultValue() default NO_DEFAULT;
+
+    /**
+     * Specify the default value documentation, when it cannot be expressed by @{link #defaultValue()}.
```

Comment:
A small example like the Hibernate one could help usability I think. But otherwise that looks good.

Reply:
Done.